x/stakeibc/keeper/msg_server_redeem_stake.go

```diff
@@ -48,15 +48,22 @@ func (k msgServer) RedeemStake(goCtx context.Context, msg *types.MsgRedeemStake)
 		return nil, sdkerrors.Wrapf(sdkerrors.ErrInvalidAddress, "invalid receiver address (%s)", err)
 	}
 
-	if msg.Amount > hostZone.StakedBal {
-		return nil, sdkerrors.Wrapf(types.ErrInvalidAmount, "cannot unstake an amount g.t. staked balance on host zone: %d", msg.Amount)
-	}
-
+	// construct desired unstaking amount from host zone
 	amt, err := cast.ToInt64E(msg.Amount)
 	if err != nil {
 		k.Logger(ctx).Error(fmt.Sprintf("error casting RedeemStake msg.Amount to int64, err: %s", err.Error()))
 		return nil, sdkerrors.Wrapf(types.ErrInvalidAmount, fmt.Sprintf("invalid amount: %s", err.Error()))
 	}
+	coinDenom := types.StAssetDenomFromHostZoneDenom(hostZone.HostDenom)
```

Comment:
nit: should we rename this to `stDenom` ?

Reply:
Done